tests/python/rest_api/test_jobs.py

```diff
@@ -605,3 +606,47 @@ def test_exported_job_dataset_structure(
             )  # images + annotation file
             content = zip_file.read(anno_file_name)
         check_func(content, values_to_be_checked)
+
+
+@pytest.mark.usefixtures("restore_db_per_class")
+class TestGetJobPreview:
+    def _test_get_job_preview_200(self, username, jid, **kwargs):
+        with make_api_client(username) as client:
+            (_, response) = client.jobs_api.retrieve_preview(jid, **kwargs)
+            assert response.status == HTTPStatus.OK
+            assert Image.open(BytesIO(response.data)).size != 0
```

Comment:
Can we compare the actual content of the preview?

Reply:
I don't think this can be easily implemented: we have several types of input data, a lot of task settings (start frame, segment size, etc.), so we need to completely reuse all the logic of the task creation process to get absolutely the same same pipeline as backend has. Probably we can get first frame of task, job or project by corresponding backend request, use PIL.Image.thumbnail and compare results. What do you think?